zebra-network/src/peer/handshake.rs

```diff
@@ -474,19 +522,28 @@ where
                                     span: tracing::Span::current(),
                                 }) {
                                     Ok(()) => {
-                                        match server_tx.flush().await {
-                                            Ok(()) => {}
+                                        // TODO: also wait on the shutdown_rx here
+                                        match timeout(
```

Comment:
turn timeout match into a generic function that logs traces and returns a BoxErr result using Result::flatten?

Reply:
Part of #2009